src/core/sock/sockinfo_tcp.cpp

```diff
@@ -2762,7 +2774,27 @@ int sockinfo_tcp::bind(const sockaddr *__addr, socklen_t __addrlen)
         }
     } else {
         si_tcp_logdbg("OS bind to %s", sockaddr2str(__addr, __addrlen, true).c_str());
-        ret = orig_os_api.bind(m_fd, __addr, __addrlen);
+        if (INPORT_ANY == in_port && m_bind_no_port) {
```

Comment:
Please move the bind_no_port logic to a separate method

Reply:
ok. and i will also move it before the line:
if (INPORT_ANY == in_port && (m_pcb.so_options & SOF_REUSEADDR)) {

because we dont care if reuseaddr was set or not.